test/unit/mem/buf/src/ConstBufTest.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2025 Anton Reinhard
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#include <alpaka/mem/buf/Traits.hpp>
+#include <alpaka/test/Extent.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/mem/view/ViewTest.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+
+#include <catch2/catch_message.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include <numeric>
+#include <type_traits>
+
+namespace buftest
+{
+    template<typename TDev, typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocConstBuf(TDev dev, TExtent extent) -> alpaka::ConstBuf<TDev, TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+
+    template<typename TBuf>
+    concept onlyConstNativePtr = requires(TBuf t) {
+        {
+            alpaka::getPtrNative(t)
+        } -> std::same_as<alpaka::Elem<TBuf> const*>;
+    };
+
+} // namespace buftest
+
+template<typename TAcc>
+static auto testConstBuffer(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<TAcc>> const& extent) -> void
```

Comment:
Can you please add also an test, which checks the return type of the access operator of a const and non const buffer. Therefor you need to allocate an 1D buffer.

```c++
static_assert(std::is_same_v<decltype(buf[0]), int&>);
```